catroid/src/org/catrobat/catroid/ui/dialogs/OrientationDialog.java

```diff
@@ -143,4 +122,47 @@ public void setCreateEmptyProject(boolean isChecked) {
 	public void setCreateDroneProject(boolean isChecked) {
 		createDroneProject = isChecked;
 	}
+
+    private class LoadingOnCreatingProject extends AsyncTask<String, Void, Void> {
+        private ProgressDialog progressDialog;
+
+		protected LoadingOnCreatingProject() {
+			progressDialog = new ProgressDialog(getActivity());
+		}
+
+		@Override
+        protected void onPostExecute(Void result) {
+            Intent intent = new Intent(getActivity(), ProjectActivity.class);
+            intent.putExtra(Constants.PROJECTNAME_TO_LOAD, projectName);
+
+            if (isOpenedFromProjectList()) {
+                intent.putExtra(Constants.PROJECT_OPENED_FROM_PROJECTS_LIST, true);
+            }
+
+            getActivity().startActivity(intent);
+            dismiss();
+            progressDialog.dismiss();
+        }
+
+		@Override
+		protected void onPreExecute() {
+			this.progressDialog.setMessage(getResources().getString(R.string.please_wait));
+			this.progressDialog.show();
+		}
+
+		@Override
+        protected Void doInBackground(String... params) {
+
+            try {
+                ProjectManager.getInstance().initializeNewProject(projectName, getActivity(), createEmptyProject, createDroneProject, createLandscapeProject);
+            } catch (IllegalArgumentException illegalArgumentException) {
+					Utils.showErrorDialog(getActivity(), R.string.error_project_exists);
+				} catch (IOException ioException) {
+					Utils.showErrorDialog(getActivity(), R.string.error_new_project);
+					Log.e(TAG, Log.getStackTraceString(ioException));
+					dismiss();
```

Comment:
is the `onPostExecute` executed after the dismiss call? if yes, i sense problems in case of an exception